sphinx/application.py

```diff
@@ -1178,6 +1178,15 @@ def add_autodoc_attrgetter(self, typ: type, getter: Callable[[Any, str, Any], An
         logger.debug('[app] adding autodoc attrgetter: %r', (typ, getter))
         self.registry.add_autodoc_attrgetter(typ, getter)
 
+    def add_linkcode_domain(self, domain: str, keys: list[str]) -> None:
+        """Register a new list of keys to use for a domain by the
+        :mod:`sphinx.ext.linkcode` extension.
+
+        .. versionadded:: 7.3
+        """
+        from sphinx.ext.linkcode import domain_keys
+        domain_keys[domain] = keys
```

Comment:
Just wondering but what happens if two extensions do the same? should extend the keys and removing duplicates? 

Also, you should probably say (in the doc) that the keys an extension defines overrides whatever keys were previously defined if we go with the current implementation.

Reply:
> Just wondering but what happens if two extensions do the same? should extend the keys and removing duplicates?

I don't think so, the idea is that an extension that provides support for a new domain, for instance `sphinxcontrib-phpdomain`, can now add the `php` domain to the list of domains supported by `linkcode`. I don't think that anybody uses two extensions adding support for the same domain, so I don't see any chances of conflict here.

Another possibility I see would be to completely forbid overriding an existing key, or maybe add a warning when a key is overridden. But I thought overriding the existing keys could also be useful in a way.

Maybe we should wait for another opinion? I'm not in a hurry for this feature.

> Also, you should probably say (in the doc) that the keys an extension defines overrides whatever keys were previously defined if we go with the current implementation.

I will add a more descriptive explanation when we settle on the preferred behaviour.

Reply:
I would like the opinion of another maintainer/contributor. Since we are in a holiday period, it might take a while though. When we release new versions, we create an issue and you can ask at that moment if we didn't decide on anything.